docs: add new client conversion - java (#898)
EventStore/documentation: docs/dev-center/tutorials/eventstore-to-kurrent-migration.md

## Why Migrate?

- **Latest features and improvements**: The KurrentDB client includes performance enhancements and new capabilities
- **Ongoing support**: Future updates and bug fixes will only be available in the KurrentDB client
- **Automated migration**: Use [OpenRewrite](https://docs.openrewrite.org) to handle the migration automatically with minimal manual effort

## What Gets Updated

This migration will automatically update:
- Maven/Gradle dependencies
- Package imports (`com.eventstore.dbclient` → `io.kurrent.dbclient`)
- Class names (`EventStoreDBClient` → `KurrentDBClient`)
- Connection strings (`esdb://` → `kurrent://`)
- Method names (`expectedRevision()` → `streamRevision()`)

## Requirements
- Java 8 or above
- Maven or Gradle (if not using the Maven or Gradle wrappers)
- A Java application that uses `com.eventstore.db-client-java` dependency.

## Setup

> First, create a composite recipe comprising of soem recipes that will change the code and dependencies.

Reviewer: I'm not sure what you are saying here.  "Some" is misspelled, and "recipe" shouldn't be used twice.

      oldFullyQualifiedTypeName: com.eventstore.dbclient.EventStoreDBProjectionManagementClient
      newFullyQualifiedTypeName: io.kurrent.dbclient.KurrentDBProjectionManagementClient
  - org.openrewrite.java.ChangeType:
      oldFullyQualifiedTypeName: com.eventstore.dbclient.EventStoreDBPersistentSubscriptionsClient
      newFullyQualifiedTypeName: io.kurrent.dbclient.KurrentDBPersistentSubscriptionsClient
  - org.openrewrite.java.ChangeType:
      oldFullyQualifiedTypeName: com.eventstore.dbclient.ExpectedRevision
      newFullyQualifiedTypeName: io.kurrent.dbclient.StreamState
  - org.openrewrite.java.ChangePackage:
      oldPackageName: com.eventstore.dbclient
      newPackageName: io.kurrent.dbclient
  - org.openrewrite.text.FindAndReplace:
      find: expectedRevision(
      replace: streamRevision(
      caseSensitive: true
  - org.openrewrite.text.FindAndReplace:
      find: esdb://
      replace: kurrent://
      caseSensitive: true
  - org.openrewrite.text.FindAndReplace:
      find: esdb+discover://
      replace: kurrent+discover://
      caseSensitive: true
---

```

This file defines a new OpenRewrite recipe named `io.kurrent.java.UpgradeClient`. 

## Execution

> Depending on which build automation tool you use (Maven or Gradle), the execution of the above composite recipe is slightly different:

Reviewer: ```suggestion
The execution of the recipe varies slightly depending on the build automation tool used (Maven or Gradle).
```

[WARNING]     org.openrewrite.text.FindAndReplace: {find=esdb://, replace=kurrent://, caseSensitive=true}
[WARNING]         org.openrewrite.text.FindAndReplace: {find=esdb+discover://, replace=kurrent+discover://, caseSensitive=true}
[WARNING] Changes have been made to LoanApplicationDemo/Java/loan-approvals/cluster-config.yml by:
[WARNING]     org.openrewrite.text.FindAndReplace: {find=esdb+discover://, replace=kurrent+discover://, caseSensitive=true}
[WARNING] Changes have been made to LoanApplicationDemo/Java/loan-approvals/config.yml by:
[WARNING]     org.openrewrite.text.FindAndReplace: {find=esdb+discover://, replace=kurrent+discover://, caseSensitive=true}
[WARNING] Changes have been made to LoanApplicationDemo/Java/loan-approvals/shared.yml by:
[WARNING]     org.openrewrite.text.FindAndReplace: {find=esdb+discover://, replace=kurrent+discover://, caseSensitive=true}
[WARNING] Changes have been made to LoanApplicationDemo/Java/loan-approvals/README.md by:
[WARNING]     org.openrewrite.text.FindAndReplace: {find=esdb+discover://, replace=kurrent+discover://, caseSensitive=true}
[WARNING] Please review and commit the results.
[WARNING] Estimate time saved: 1h 20m

```

> Once the recipe has been executed, package names, class names, dependencies and connection strings, would have been updated to use the KurrentDB Java client.

Reviewer: see:  https://github.com/kurrent-io/documentation/blob/tutorials-framework/docs/resources/guides/Java_Client_Rebrand.md where David added a broader explanation of what has transpired.  I think this is worth adding.

# Breaking changes
   ### expectedRevision

The `OptionsWithExpectedRevisionBase` has been replaced with `OptionsWithStreamStateBase`. These classes provide the ability to set the expected revision on `AppendToStreamOptions` and `DeleteStreamOptions`. 

Whilst the former class had two overloaded methods named `expectedRevision` allowing to pass in a `ExpectedRevision` object or a `long`, in the latter, the methods are named differently.

    1. streamState(StreamState state)
    2. streamRevision(long revision)

> This means we can't automatically convert without knowing the type of the parameter. 

Reviewer: ```suggestion
This means you can't automatically convert without knowing the type of the parameter. 
```
